device/tt_silicon_driver.cpp

```diff
@@ -2957,3 +2959,64 @@ tt_version tt_SiliconDevice::get_ethernet_fw_version() const {
     log_assert(eth_fw_version.major != 0xffff and eth_fw_version.minor != 0xff and eth_fw_version.patch != 0xff, "Device must be started before querying Ethernet FW version.");
     return eth_fw_version;
 }
+
+// v1 functions
+void tt_SiliconDevice::write_to_device(const void *mem_ptr, uint32_t size_in_bytes, chip_id_t chip, CoreCoord_V1 core_coord, uint64_t addr, const std::string& tlb_to_use) {
+    CoreCoord_V1 physical_core_coord = coordinate_manager->to_physical(core_coord);
+
+
+}
+
+void tt_SiliconDevice::read_from_device(void* mem_ptr, chip_id_t chip, CoreCoord_V1 core_coord, uint64_t addr, uint32_t size, const std::string& fallback_tlb) {
+    CoreCoord_V1 physical_core_coord = coordinate_manager->to_physical(core_coord);
+
+
+}
+
+// v2 functions
+
+// tensix core coord
+void tt_SiliconDevice::write_to_device(const void *mem_ptr, uint32_t size_in_bytes, chip_id_t chip, TensixCoreCoord_V2 core_coord, uint64_t addr) {
```

Comment:
In any case we will probably end up with write_to_device functions per coreType. 
I do like how V1 will hide this though, probably by having a switch case statement pointing to those functions.

Reply:
Yes we can discuss about it in more detail later on, but one function can do the job as well. If we translate coordinates to physical, we can just program the TLB and do the write, it will work as well